src/kvstore/comm.h

```diff
@@ -79,8 +80,18 @@ class Comm {
     return pinned_ctx_;
   }
 
+  /**
+   * \brief Sets gradient compression parameters to be able to
+   * perform reduce with compressed gradients
+   */
+  void SetGradientCompression(Gc* gc) {
```

Comment:
Passing raw pointers like this is really bad

Reply:
Use a shared ptr

Reply:
changed